src/main/java/rx/internal/util/SubscriptionList.java

```diff
@@ -34,7 +29,6 @@
 
     private LinkedList<Subscription> subscriptions;
     private volatile boolean unsubscribed;
-    private final ReentrantLock lock = new ReentrantLock();
```

Comment:
I've benchmarked this via OperatorPublishPerf and using ```synchronized``` instead of lock gave about 25% more throughput with 1M elements.